### PR TITLE
Non-functional improvements in event-publisher-proxy (documentation, testing)

### PR DESCRIPTION
**Pull request by nachtmaar**
**Description**

The following changes have been performed while reviewing some code in the [event-publisher-proxy](https://github.com/kyma-project/kyma/pull/13120/commits).

This PR adds several non-functional improvements such as:

- Some `documentation`
- Using functional options pattern for `pkgnats.Connection`
- Fix goroutine loop bug in some parallel tests (see [here](https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721) for more explanation)

About the failing optional eventing tests:
- kyma-gardener-gcp-eventing: https://github.com/kyma-project/kyma/issues/13576
- kyma-skr-eventing: https://github.com/kyma-project/kyma/issues/13393

**Related issue(s)**

### CODE REVIEW

**Comment by nachtmaar** [comment]
Blocked by https://github.com/kyma-project/kyma/pull/13120

File: components/event-publisher-proxy/pkg/nats/connect.go [hunk 1]
```diff
@@ -8,33 +8,65 @@ import (
...
+	return bc
 }
 
-// Connect returns a nats connection that is ready for use, or error if connection to the nats server failed
-func (bc *BackendConnection) Connect() error {
-	connection, err := nats.Connect(bc.connectionData.url, nats.RetryOnFailedConnect(bc.connectionData.retry),
-		nats.MaxReconnects(bc.connectionData.reconnects), nats.ReconnectWait(bc.connectionData.wait))
+// Connect returns a NATS connection that is ready for use, or an error if connection to the NATS server failed.
+// It uses the nats.Connect function which is thread-safe.
+func (bc *Connection) Connect() error {
```
Comment (k15r) [inline components/event-publisher-proxy/pkg/nats/connect.go:66]: please consider changing `bc` to `c` to stay consistent

File: components/event-publisher-proxy/pkg/sender/nats.go [hunk 2]
```diff
@@ -18,48 +18,45 @@ type GenericSender interface {
...
 	ctx               context.Context
 	logger            *logrus.Logger
-	backendConnection *pkgnats.BackendConnection
+	backendConnection *pkgnats.Connection
 }
 
-// NewNatsMessageSender returns a new NewNatsMessageSender instance with the given nats connection.
-func NewNatsMessageSender(ctx context.Context, bc *pkgnats.BackendConnection, logger *logrus.Logger) *NatsMessageSender {
+// NewNatsMessageSender returns a new NewNatsMessageSender instance with the given NATS connection.
+func NewNatsMessageSender(ctx context.Context, bc *pkgnats.Connection, logger *logrus.Logger) *NatsMessageSender {
```
Comment (k15r) [inline components/event-publisher-proxy/pkg/sender/nats.go:29]: would it be possible to rename `pkgnats` to `nats` or something similar

**Comment by marcobebway** [comment]
Heads-up, the approach used in https://github.com/kyma-project/kyma/pull/13120 might change, so please do not merge this PR until https://github.com/kyma-project/kyma/pull/13120 is merged.

File: components/event-publisher-proxy/pkg/sender/nats.go (hunk 2 above)
Comment (nachtmaar) [reply components/event-publisher-proxy/pkg/sender/nats.go]: In theory yes. We do import 2 packages in this file which are called `nats`.
Both are using a custom import name. 

```go
	cenats "github.com/cloudevents/sdk-go/protocol/nats/v2"
	pkgnats "github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/nats"
```

I would agree renaming it if we would not have a collision on the import names. However I am not sure why we should rename to `pkgnats` and not `cenats` to `nats` then. I like the current version where both imports use an explicit name.

What do you think ?

**Review by FriedrichWilken — APPROVED** [review]

**Pull request merged**